bpdm-pool/src/main/kotlin/org/eclipse/tractusx/bpdm/pool/repository/LegalEntityRepository.kt

```diff
@@ -37,6 +38,28 @@ interface LegalEntityRepository : PagingAndSortingRepository<LegalEntity, Long>,
 
     fun findByUpdatedAtAfter(updatedAt: Instant, pageable: Pageable): Page<LegalEntity>
 
+    /*@Query("SELECT DISTINCT p FROM LegalEntity p WHERE p.legalName.value = :value")
+    fun findByLegalNameValue(value:String, pageable: Pageable): Page<LegalEntity>*/
+
+    /*@Query("SELECT DISTINCT p, levenshtein_distance(p.legalName.value, :value) AS distance\n" +
+            "FROM LegalEntity p\n" +
+            "WHERE p.legalName.value LIKE :value\n" +
+            "ORDER BY distance DESC\n")
+    fun findByLegalNameValue(value: String, pageable: Pageable): Page<LegalEntity>*/
+
+    /*@Query(value = "SELECT p.*, SIMILARITY(p.legalName.value, :value) AS score " +
+            "FROM LegalEntity p " +
+            "WHERE SIMILARITY(p.legalName.value, :value) > 0.3 " + // Adjust the similarity threshold as needed
+            "ORDER BY score DESC",
+        countQuery = "SELECT COUNT(*) FROM LegalEntity p WHERE SIMILARITY(p.legalName.value, :value) > 0.3", // Count query
+        nativeQuery = true)
+    fun findByLegalNameValue(value:String, pageable: Pageable): Page<LegalEntity>*/
+
+    /*fun findByLegalNameValue(value: String, pageable: Pageable): Page<LegalEntity>*/
+
+    @Query("SELECT DISTINCT p FROM LegalEntity p WHERE LOWER(p.legalName.value) LIKE LOWER(CONCAT('%', :value, '%'))")
```

Comment:
Maybe order by Length of name so that possible exact or close matches appear first? Just an idea

Reply:
Adapted with approach, In this modified query, now it first checks for:
1. An exact match (with a sorting value of 0) using the CASE statement.
2. Close matches (with a sorting value of 1) where the length of legalName.value is less than or equal to the length of :value.
3. Other matches (with a sorting value of 2).